allennlp/data/vocabulary.py

```diff
@@ -455,6 +456,29 @@ def extend_from_vocab(self, vocab: "Vocabulary") -> None:
             for token in vocab.get_token_to_index_vocabulary(namespace):
                 self.add_token_to_namespace(token, namespace)
 
+    def extend_from_dictionary(
+        self,
+        encoding_dictionary: Dict[str, int],
+        namespace: str = "from_transformers",
+        resave_to_files: bool = False,
```

Comment:
This is getting to be a bigger change, but it's probably necessary to really fix this the right way.  But I *think* this function should take arguments for setting padding and oov tokens, and those should set them in a namespace-specific way.  And the padding and oov tokens then would need to be serialized with the vocabulary and read from the serialized files.  Then we wouldn't need the `from_transformers` namespace added to the non-padded namespaces.

I would wait to do any work on this, though, until @dirkgr has had a chance to glance at it and see if he agrees.  I think this is the right approach, but I could be missing something.